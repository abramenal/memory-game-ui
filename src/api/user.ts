import { API_BASE_URL } from '../constants';
import { GameTurn, User } from '../types';

type LoginParams = {
  username: string;
}

export const login = ({ username }: LoginParams): Promise<User> => fetch(
  `${API_BASE_URL}/users/login`,
  {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      username,
    }),
  },
).then((res) => res.json());

type GetHistoryParams = {
  userId: string;
}

export const getUserHistory = ({ userId }: GetHistoryParams): Promise<GameTurn[]> => fetch(
  `${API_BASE_URL}/users/${userId}/history`,
  {
    method: 'GET',
  },
).then((res) => res.json());
