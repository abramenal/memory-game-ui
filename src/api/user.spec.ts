import { login, getUserHistory } from './user';

describe('login', () => {
  beforeAll(() => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
    // @ts-ignore
    globalThis.fetch = jest.fn((...args) =>
      Promise.resolve({
        json: () => args,
      }),
    );
  });

  it('calls HTTP API', async () => {
    const request = await login({ username: 'test' });
    expect(request).toMatchSnapshot();
  });
});

describe('getUserHistory', () => {
  beforeAll(() => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
    // @ts-ignore
    globalThis.fetch = jest.fn((...args) =>
      Promise.resolve({
        json: () => args,
      }),
    );
  });

  it('calls HTTP API', async () => {
    const request = await getUserHistory({ userId: 'test' });
    expect(request).toMatchSnapshot();
  });
});
