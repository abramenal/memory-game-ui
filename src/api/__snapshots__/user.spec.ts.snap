// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUserHistory calls HTTP API 1`] = `
Array [
  "http://localhost:3000/users/test/history",
  Object {
    "method": "GET",
    "mode": "cors",
  },
]
`;

exports[`login calls HTTP API 1`] = `
Array [
  "http://localhost:3000/users/login",
  Object {
    "body": "{\\"username\\":\\"test\\"}",
    "headers": Object {
      "Content-Type": "application/json",
    },
    "method": "POST",
    "mode": "cors",
  },
]
`;
